test/BaseTabs.test.ts

```diff
@@ -0,0 +1,42 @@
+import { test, expect } from 'vitest';
+import { render } from '@testing-library/vue';
+import BaseTabs from '../src/shared/ui/components/BaseTabs.vue';
+import type { DefaultTransactionTypeTabs } from '../src/features/filter-transactions/model';
+import { defaultTransactionTypeOptions } from '../src/features/filter-transactions/model';
+import { ref } from 'vue';
+import { adaptiveTransactionTypeOptions } from '../src/pages/Accounts/consts';
+import { applyAdaptiveOptions } from '../src/shared/utils/adaptive-options';
+import { i18n } from '../src/shared/lib/localization';
+
+test('BaseTabs adaptive display correctness', async () => {
+  const model = ref<DefaultTransactionTypeTabs>('transactions');
+
+  // Since rerender method doesn't work as expected due to changing window width
+  // we will render & unmount our component every time
+  function renderBaseTabs(width: number) {
+    window.innerWidth = width;
+
+    const { unmount } = render(BaseTabs, {
+      global: {
+        plugins: [i18n],
+      },
+      props: {
+        items: defaultTransactionTypeOptions,
+        modelValue: model.value,
+        adaptiveOptions: adaptiveTransactionTypeOptions,
+      },
+    });
+
+    const tabs = document.getElementsByClassName('base-tabs__tab').length;
+    expect(tabs).toBe(applyAdaptiveOptions(width, adaptiveTransactionTypeOptions));
```

Comment:
This use of `applyAdaptiveOptions` as an expected value some reduces the usefulness of the test. Better to put an exact expected number - test would also become more expressive.